### ISSUE TITLE
Page with show_on_menu selected breaks Reorder Nav

### ISSUE THREAD

**Issue by erickgonzalez** [dotCMS/core#22137]
**Describe the bug**
If a page under a folder has `show_on_menu` selected and the reorder code of the nav is added to that page, when trying to reorder an exception is thrown

```
[02/05/22 18:35:38:706 UTC] WARN action.RequestProcessor: Unhandled Exception thrown: class java.lang.ClassCastException
[02/05/22 18:35:38:706 UTC] ERROR portlet.StrutsPortlet: java.lang.ClassCastException: class com.dotmarketing.portlets.contentlet.model.Contentlet cannot be cast to class com.dotmarketing.beans.WebAsset (com.dotmarketing.portlets.contentlet.model.Contentlet and com.dotmarketing.beans.WebAsset are in unnamed module of loader org.apache.catalina.loader.ParallelWebappClassLoader @37ff4054)
[02/05/22 18:35:38:709 UTC] ERROR servlet.MainServlet: com.dotcms.repackage.javax.portlet.PortletException
```

**To Reproduce**
1. Create a new folder named `4-h-money-smarts`
2. Create a couple of folder inside it and check Show On Menu.
3. Create a page named `index` and check Show On Menu.
4. Upload the following vtl's to the folder (change the extension to vtl):
[nav-arrange.txt](https://github.com/dotCMS/core/files/8611896/nav-arrange.txt)
[navigation-desktop.txt](https://github.com/dotCMS/core/files/8611897/navigation-desktop.txt)
5. Add a Widget to the page with the following code:
```
#dotParse("/4-h-money-smarts/nav-arrange.vtl")
#dotParse("/4-h-money-smarts/navigation-desktop.vtl")
```
6. Click over the `Rearrange Navigation` link.

**Expected**
We should be able to reorder navigation, for the folders that's how always worked, successfully even though a page has Show On Menu.

Reproducible in demo(22.03)
Customer ticket: https://dotcms.zendesk.com/agent/tickets/107024

**Comment by nollymar** [comment]
PR: #22181

**Comment by fabrizzio-dotCMS** [comment]
Looks good to me

**Comment by bryanboza** [comment]
Fixed, tested on release-22.06 // Docker // FF

**Comment by erickgonzalez** [comment]
Not reproducible in 21.06 or 5.3.8 LTS.
